workflows/flows/assemble_study_tasks/make_samplesheets.py

```diff
@@ -123,48 +123,34 @@ def make_samplesheet(
                     }
                     else (
                         analyses.models.Assembler.SPADES
-                        if metadata.get(
-                            analyses.models.Run.CommonMetadataKeys.INFERRED_LIBRARY_LAYOUT,
-                            metadata.get(
-                                analyses.models.Run.CommonMetadataKeys.LIBRARY_LAYOUT
-                            ),
+                        if assembly.run.inferred_metadata.get(
+                            analyses.models.Run.CommonMetadataKeys.LIBRARY_LAYOUT
                         )
                         == SINGLE_END_LIBRARY_LAYOUT
-                        and metadata.get(
+                        and assembly.run.inferred_metadata.get(
```

Comment:
Maybe I understood not whole code correctly.. but my first question was: how I should know does field have `inferred` record or not? We know in advance that we have implemented inferred for `library_layout` but if you will add it to `library_strategy`, for example, how I should find/check it? 
In my understanding inferred function should work in a bit different way: I should use original model name and then model fuction should decide: use inferred value (if that exists) or original. So I will not need to check does `inferred` exists of not. 
I'm not sure it does make sense to implement in Django terms.. but maybe something to think about

Reply:
Thanks @KateSakharova . This does indeed work how you wish it to, in that `inferred_metadata` is a sort of clever version of `metadata`, but when a key within `metadata` starts with `inferred_` then that it preferred to one that does not.
A bit hard to follow in the implementation code I think, but in the tests it is clearer: https://github.com/EBI-Metagenomics/emgapi-v2/pull/91/files#diff-364fcc141694a145044a135f232afceaa22c852cb1f62a242b6ee4e7235213d7R460

I considered overriding the `.metadata` itself, but that seemed a bit dangerously magical and likely to break something in django internals potentially.

But yes, as it stands you can just do `model.inferred_metadata["something"]` and not care about whether we have `inferred_something` or not.

Reply:
Good point :) I updated README with details of a bunch of the helper things we have (ones that I could remember anyway)